sdk/src/tx/txParamProcessor.ts

```diff
@@ -0,0 +1,111 @@
+import { AddressLookupTableAccount, Connection, RpcResponseAndContext, SimulatedTransactionResponse, Transaction, TransactionInstruction, TransactionVersion, VersionedTransaction } from "@solana/web3.js";
+import { BaseTxParams, ProcessingTxParams } from "..";
+
+const COMPUTE_UNIT_BUFFER_FACTOR = 1.2;
+
+const TEST_SIMS_ALWAYS_FAIL = false;
+
+type TransactionProps = {
+    instructions: TransactionInstruction | TransactionInstruction[],
+    txParams?: BaseTxParams,
+    txVersion?: TransactionVersion,
+    lookupTables?: AddressLookupTableAccount[]
+}
+
+/**
+ * This class is responsible for running through a "processing" pipeline for a base transaction, to adjust the standard transaction parameters based on a given configuration.
+ */
+export class TransactionProcessor {
+
+    private static async getComputeUnitsFromSim (
+		txSim: RpcResponseAndContext<SimulatedTransactionResponse>
+	) {
+		if (txSim?.value?.unitsConsumed) {
+			return txSim?.value?.unitsConsumed * 1.2;
+		} 
+
+		return undefined;
+	}
+
+	private static async getTxSimComputeUnits(
+		tx : VersionedTransaction,
+        connection: Connection
+	) {
+		try {
+
+            if (TEST_SIMS_ALWAYS_FAIL) throw new Error('Test Error::SIMS_ALWAYS_FAIL');
+
+			const simTxResult = (await connection.simulateTransaction(
```

Comment:
how do we wanna handle an error in simulation?

Reply:
At the moment it'll basically ignore the error, and will use whatever default compute units were passed in with the other txParams .. the implication is that we need to remember to give a fallback number of compute units for all transactions ... imo I thought this was good - what else do you have in mind for error handling?

My heuristic is something like : Simulations are probs more expensive on the RPC, therefore more likely to fall over for some reason, but we want to continue trying the transaction anyway without flashing an error to the user which might be confusing (and also not really something they'd be able to fix?)